Makefile

```diff
@@ -117,6 +117,7 @@ GENERATE_YAML_ENV_VARS := -e CLUSTER_NAME -e SERVICE_CIDR -e CLUSTER_CIDR -e CAP
 dev-yaml: | clusterctl-tools
 	docker run -v $(CWD):/go/src/sigs.k8s.io/cluster-api-provider-vsphere \
 	  -w /go/src/sigs.k8s.io/cluster-api-provider-vsphere \
+	  -u $(shell id -u):$(shell id -g) \
```

Comment:
Hi @andrewsykim,

Every use of `shell` starts a new sub-process. Please either:

1. Define both once as Makefile vars, or
2. Move the logic to the shell: `$$(id -u):$$(id -g)` so running `make` doesn't need to call these unless calling these targets